crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+                    for _ in 0..report.rotations {
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                wasm = mutated;
+                            }
+                            Err(_) => { /* bypass */ }
+                        }
+                    }
+                    println!("Reached");
+                    while report.rotations <= ROTATIONS {
+                        report.rotations += 1;
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                let mut validator = wasmparser::Validator::new();
+                                match validator.validate_all(&mutated) {
+                                    Ok(_) => {
+                                        match self.compile(&mutated, &name, report.rotations) {
+                                            Ok(mhashes) => {
+                                                let hash = self.hash(mutated.clone());
+                                                wasm = mutated;
+                                                report.add_wasm_hash(hash);
+
+                                                for (idx, mhash) in mhashes.iter().enumerate() {
+                                                    report.add_obj_hash(mhash.clone(), idx);
+                                                }
+
+                                                if report.rotations % 100 == 0 {
+                                                    // Save report
+                                                    self.save_report(&report, &name);
+                                                    if self.print {
+                                                        report.print();
+                                                    }
+                                                }
+                                            }
+                                            Err(e) => report
+                                                .errors
+                                                .push((report.rotations as u64, format!("{}", e))),
+                                        }
+                                    }
+                                    // All generated Wasm should be valid
+                                    Err(e) => panic!("{}", e),
+                                }
+                            }
+                            Err(e) => report
+                                .errors
+                                .push((report.rotations as u64, format!("{}", e))),
+                        }
+                    }
+                }
+                None => {
+                    break;
+                }
+            }
+        }
+    }
+
+    fn hash(&self, data: Vec<u8>) -> String {
+        // By default md5 is the hashes here
+        format!("{:x}", Md5::digest(&data))
```

Comment:
md5 doesn't seem like the best choice here.

Either

* we don't care a ton about using a good, wide hash to avoid collisions, and can just use the `std` library's default hasher for hash maps like

  ```rust
  use std::collections::hash_map::DefaultHasher;
  use std::hash::{Hash, Hasher};

  let mut h = DefaultHasher::default();
  data.hash(&mut h);
  h.finish()
  ```

* or we do care about using a good, wide hash to avoid collisions, in which case we want something better than md5, like [blake3](https://crates.io/crates/blake3)